src/NServiceBus.RavenDB.Tests/SagaPersister/When_persisting_a_saga_with_the_same_unique_property_as_another_saga.cs

```diff
@@ -13,37 +12,40 @@ public class When_persisting_a_saga_with_the_same_unique_property_as_another_sag
     [Test]
     public async Task It_should_enforce_uniqueness()
     {
-        IAsyncDocumentSession session;
-        var options = this.CreateContextWithAsyncSessionPresent(out session);
-        var persister = new SagaPersister();
-        var uniqueString = Guid.NewGuid().ToString();
-
-        var saga1 = new SagaData
+        using (var session = store.OpenAsyncSession().UsingOptimisticConcurrency().InContext(out var options))
         {
-            Id = Guid.NewGuid(),
-            UniqueString = uniqueString
-        };
-
-        var synchronizedSession = new RavenDBSynchronizedStorageSession(session);
-
-        await persister.Save(saga1, this.CreateMetadata<SomeSaga>(saga1), synchronizedSession, options);
-        await session.SaveChangesAsync().ConfigureAwait(false);
-        session.Dispose();
+            var persister = new SagaPersister();
+            var uniqueString = Guid.NewGuid().ToString();
 
-        var exception = await Catch<ConcurrencyException>(async () =>
-        {
-            options = this.CreateContextWithAsyncSessionPresent(out session);
-            synchronizedSession = new RavenDBSynchronizedStorageSession(session);
-            var saga2 = new SagaData
+            var saga1 = new SagaData
             {
                 Id = Guid.NewGuid(),
                 UniqueString = uniqueString
             };
-            await persister.Save(saga2, this.CreateMetadata<SomeSaga>(saga2), synchronizedSession, options);
+
+            var synchronizedSession = new RavenDBSynchronizedStorageSession(session);
+
+            await persister.Save(saga1, this.CreateMetadata<SomeSaga>(saga1), synchronizedSession, options);
             await session.SaveChangesAsync().ConfigureAwait(false);
-        });
+            session.Dispose();
```

Comment:
Why was this left here? If I understand this test correctly the sessions don't really need to overlapp or be disposed for this test to pass right?

Reply:
Good spot. I missed this line while working on the second commit. I'll fix it and amend.